### ISSUE TITLE
Add plan and approval stage for CodePipeline

### ISSUE THREAD

**Issue by hoanglc** [aws-ia/terraform-aws-control_tower_account_factory#481]
**Describe the outcome you'd like**

When initializing the AFT product, I would love to have 2 another CodePipeline stages for each AFT repositories (account request, account customization, global customization and account provisioning customizations) 
**(1) Plan stage**
Run "terraform plan"

**(2) Approval stage**
To avoid terraform code is applied without approval.

**Is your feature request related to a problem you are currently experiencing? If so, please describe.**

Customer want to have terraform plan output to detect any change if had. and also approval stage on CodePipeline to have double-check if something went wrong.

**Additional context**

CodePipeline with 2 stages. Wait for 2 additional stage
<img width="1344" alt="CodePipeline with 2 stages" src="https://github.com/user-attachments/assets/9d604f66-ba26-4fc4-90da-c3ec199af0df">

**Comment by rbhagyar** [comment]
Requesting this feature too. Right now looking at workarounds by modifying this through customizations.

**Comment by trevorndodds** [comment]
Anyone modified their pipeline to add a plan/approval stage?

**Comment by v-rosa** [comment]
@trevorndodds Yes! Finally! Take a look https://github.com/aws-ia/terraform-aws-control_tower_account_factory/issues/636
